### ISSUE TITLE
FAB button not appearing, crashes on clicking Posts Manager

### ISSUE THREAD

**Issue by WilyBox** [oktadeveloper/okta-react-node-example#3]
I'm quite new to this and I'm not sure what the next step is in fixing things. 

I ran the example code after following the instructions. I can log in fine but the fab button won't appear and when I click Posts Manager it gives a TypeError: Cannot read property 'length' of undefined. 

I followed the previous issue but following the suggested solution (adding sqlite3) didn't fix it. Any help would be appreciated. 

![Screenshot from 2019-06-26 01-21-25](https://user-images.githubusercontent.com/40214883/60142278-fa3f4700-97b0-11e9-86fb-ceba441a8c57.png)

![Screenshot from 2019-06-26 01-26-12](https://user-images.githubusercontent.com/40214883/60142377-7174db00-97b1-11e9-80d1-d22b67740aa2.png)

package.json:

  "name": "okta-node-react-example",
  "version": "0.1.0",
  "private": true,
  "dependencies": {
    "@babel/runtime": "7.0.0-beta.55",
    "@material-ui/core": "^1.3.1",
    "@material-ui/icons": "^1.1.0",
    "@okta/jwt-verifier": "^0.0.12",
    "@okta/okta-react": "^1.0.2",
    "body-parser": "^1.18.3",
    "cors": "^2.8.4",
    "dotenv": "^6.0.0",
    "epilogue": "^0.7.1",
    "express": "^4.16.3",
    "final-form": "^4.8.3",
    "lodash": "^4.17.10",
    "moment": "^2.22.2",
    "react": "^16.4.1",
    "react-dom": "^16.4.1",
    "react-final-form": "^3.6.4",
    "react-router-dom": "^4.3.1",
    "react-scripts": "1.1.4",
    "recompose": "^0.27.1",
    "sequelize": "^4.38.0",
    "sqlite3": "^4.0.9"
  },
  "scripts": {
    "start": "npm-run-all --parallel watch:server start:web",
    "start:web": "react-scripts start",
    "start:server": "node src/server",
    "watch:server": "nodemon --watch src/server src/server",
    "build": "react-scripts build",
    "test": "react-scripts test --env=jsdom",
    "eject": "react-scripts eject"
  },
  "devDependencies": {
    "nodemon": "^1.18.8",
    "npm-run-all": "^4.1.3"
  }
}

**Comment by redbmk** [comment]
@WilyBox Since this keeps cropping up, I just pushed a change that will prevent the page from crashing and show a friendly error message instead. You should only see the FAB on the Posts Manager page, but that page has been crashing.

It seems like this happens whenever it can't fetch the posts for some reason. Could you please check that you have sqlite installed on your machine? The `sqlite3` npm module is needed in order to communicate with sqlite, but you still need to have it installed on your machine. Please try running `which sqlite3` and let me know what comes back. If it's blank, you'll need to install it (with e.g. `apt-get` or `brew` depending on your system). If you do have something like `/usr/bin/sqlite3`, then could you please look at the error messages in the browser console and paste them here?

**Comment by WilyBox** [comment]
@redbmk Hi, thanks for the reply. I have sqlite3 installed in miniconda's bin. 
The terminal reads
"Error while resolving signing key for kid "nZegQMStNhBdfoj6VgEkhpUQ5B3okfqZ7YC4EH2s6sc".
The browser is saying:
"Unhandled Rejection (TypeError): Cannot read property 'length' of undefined"
in:
PostsManager.render
line 103: {this.state.posts.length > 0 ? (

**Comment by redbmk** [comment]
@WilyBox It looks like the `OktaJwtVerifier` in `src/server/index.js` might not be getting configured properly. Do you have the line `require('dotenv').config({ path: '.env.local' });` at the top of that file?

**Comment by WilyBox** [comment]
@redbmk thanks for the quick reply, yes I have

`require('dotenv').config({ path: '.env.local' });
`

I was thinking maybe the issue is because Sequelize is using sqlite (and not sqlite3) as its dialect?

`
const database = new Sequelize({
  dialect: 'sqlite',
  storage: './test.sqlite',
});
`

**Comment by redbmk** [comment]
@WilyBox Hmm, bummer I thought that would be it. I tried commenting out that line in my code and got the same error about resolving the signing key. That error is thrown from `@okta/jwt-verifier`. Could you try `console.log(process.env)` and make sure you see `REACT_APP_OKTA_CLIENT_ID` and `REACT_APP_OKTA_ORG_URL` in there and they're what you expect (please don't post the actual values here)?

I noticed your `package.json` is slightly different than the one in this repo. Did you follow the tutorial rather than cloning this code? I'm curious if you clone this code in a separate directory, run `yarn`, and copy your `.env.local` file over, do things work then or do you get the same error? If it does work, there might be a typo in the tutorial or something.

**Comment by WilyBox** [comment]
Well I followed the tutorial, then downloaded it separately - but ended changing the package JSON trying to fix it. Monkey with a hammer, here. 
So the `console.log(process.env)` isn't finding anything, and `console.log(this.process.env)` is showing `.env` as undefined. The errors have changed to CORS errors:

`
 No 'Access-Control-Allow-Origin' header is present on the requested resource. Origin 'http://localhost:3000' is therefore not allowed access. The response had HTTP status code 401.
`

If this is a hassle, it's fine - it was probably too big a project for me anyway!

**Comment by redbmk** [comment]
Did you verify that the client ID in `.env.local` is the same as in your okta application? Also, when you created your okta application, did you change it to port 3000, or leave it at the default of 8080 (or change it to something else)? You might need to run the app on a different port (e.g. `PORT=8080 yarn start`).

Looks like another check needs to be added to the frontend for when clicking the `+` fails

**Comment by WilyBox** [comment]
I did check, they were the same. I changed it to PORT=8080 and that was promising. 
I'm on the Posts Manager, its not throwing me any errors. 

The only thing is when I try and post by clicking `+`  I'm getting:
`Unexpected token < in JSON at position 0` in red at the bottom
And on the terminal, I get: 
`Error while resolving signing key for kid "nZegQMStNhBdfoj6VgEkhpUQ5B3okfqZ7YC4EH2s6sc"`

![Screenshot from 2019-06-28 23-20-39](https://user-images.githubusercontent.com/40214883/60374297-a70d0480-99fb-11e9-8417-7b01bc2cb246.png)

This could be a result of JWT tokens having a expiry of 1 hour set as default on Okta settings as desribed in the following support case:

https://support.okta.com/help/s/question/0D51Y00006CHjMz/how-can-you-adjust-the-expiration-date-of-a-jwt-token

It's quite late here, so I'll go to bed. I'll have another crack at it tomorrow. Thanks for your time.
